### ISSUE TITLE
GradientLegend

### ISSUE THREAD

**Issue by Breviel** [pyqtgraph/pyqtgraph#1591]
### Short description
GradientLegend generates the legend upside down

### Code to reproduce

```python
from PyQt5 import QtCore, QtGui
from PyQt5.QtWidgets import QWidget, QApplication
import pyqtgraph as pg
import sys

class AnalysePlot(QWidget):

    def __init__(self, parent=None):
        QWidget.__init__(self, parent)

        self.setWindowTitle('Model parameter endring')
        self.setAutoFillBackground(True)
        self.setStyleSheet("color: yellow;"
                           "background-color: rgba(0, 0, 0, 255);")

        self.view = pg.GraphicsView(self)
        self.layout = pg.GraphicsLayout(border=(100, 100, 100))
        self.view.setCentralItem(self.layout)
        self.view.show()
        self.view.resize(640, 480)
        self.layout.setSpacing(5)

        self.p1 = self.layout.addPlot(title="Parametrene")
        self.p1.setFixedWidth(620)
        self.p1.setFixedHeight(460)
        self.p1.showGrid(1, 1, 0.3)

        self.gradLeg = pg.GradientLegend(size=(25, 100), offset=(-15, 15))

        self.updateGraph()

    def updateGraph(self):
        self.p1.clear()

        startParams = [-721, 617, 0, -666, 0, 446, 2586, 0, 4, 0, 0, -210, 2722, 165, 12, 0, 588, 0, 140, -689]
        params = [-719, 616, -11, -672, -17, 447, 2605, 2, 8, -1, 0.0, -209, 2722.0, 165.0, 12.0, 0.0, 588.0, 0.0,\
                          140.0, -689.0]
        arrayDif = []

        grad = QtGui.QLinearGradient(QtCore.QPointF(0, -20), QtCore.QPointF(20, 20))
        grad.setColorAt(0.0, pg.mkColor('r'))
        grad.setColorAt(0.125, pg.mkColor('#FF9933'))
        grad.setColorAt(0.25, pg.mkColor('y'))
        grad.setColorAt(0.375, pg.mkColor('g'))
        grad.setColorAt(0.5, pg.mkColor('b'))
        grad.setColorAt(0.625, pg.mkColor('g'))
        grad.setColorAt(0.75, pg.mkColor('y'))
        grad.setColorAt(0.875, pg.mkColor('#FF9933'))
        grad.setColorAt(1.0, pg.mkColor('r'))
        self.brush = QtGui.QBrush(grad)

        self.gradLeg.setGradient(grad)
        legendDict = {
            "-5": 0.375,
            "-10": 0.25,
            "-15": 0.125,
            "-20": 0,
            "0": 0.5,
            "5": 0.625,
            "10": 0.75,
            "15": 0.875,
            "20": 1
        }
        self.gradLeg.setLabels(legendDict)

        self.p1.setTitle('Parametrene')
        self.p1.setLabel('left', 'Endring [mm]')
        self.p1.setLabel('bottom', 'Parameter')
        self.p1.setLabel('right', 'Endring [mm]')

        for i in range(len(startParams)):
            arrayDif.append(startParams[i] - params[i])

        bargraph = pg.BarGraphItem(x=range(20), height=arrayDif, width=0.6, brush=self.brush)
        self.p1.addItem(bargraph)
        self.p1.addItem(self.gradLeg)

        ax = self.p1.getAxis('bottom')
        ticks = [(0, 'p2'), (1, 'p3'), (2, 'p4'), (3, 'p8'), (4, 'p9'), (5, 'p10'), (6, 'p12'),\
                 (7, 'p13'),  (8, 'p14'), (9, 'p16'), (10, 'p17'), (11, 'p18'), (12, 'p21'),\
                 (13, 'p23'), (14, 'p24'), (15, 'p25'), (16, 'p27'), (17, 'p28'), (18, 'p29'), \
                 (19, 'p40')]
        ax.setTicks([ticks])

if __name__ == "__main__":
    app = QApplication(sys.argv)
    ui = AnalysePlot()
    ui.show()
    app.exec_()
```

### Expected behavior

Legend should be right side up

### Real behavior
Legend is generated upside down

### Tested environment(s)

 * PyQtGraph version: 0.11.1
 * Qt Python binding: PyQt5 5.15.2 Qt 5.15.2
 * Python version:  3.7
 * Operating system:  Windows 10 Pro 64bit
 * Installation method: in PyCharm

**Comment by j9ac9k** [comment]
Hi @Breviel I hope you don't mind, I edited your code a bit to make it easier to read.

**Comment by NilsNemitz** [comment]
I made an attempt to fix this, have a look at PR #1639.

By the way, I think the gradient definition needs to be 
 `grad = QtGui.QLinearGradient(QtCore.QPointF(0, -20), QtCore.QPointF(0, 20))`  (not (20,20) )
to line up with the y axis.

**Comment by Breviel** [comment]
Hey,

Thank You for clearing up how I might need to define the gradient.
Keep up the good work, it's much appreciated :)
